### PR TITLE
GM: Set new longitudinal scaling

### PR DESCRIPTION
**Pull request by nworb-cire**
**Description**

See https://github.com/commaai/opendbc/pull/951 for a description of the issue. See https://github.com/commaai/panda/pull/1905, https://github.com/commaai/opendbc/pull/1025 for submodule PRs. 

Trivial changes: scaled all values according to the formula `f(x) = x + (4096*5) - 22534` which is the new scaling defined in the DBC file.
Nontrivial change: using the above formula, `f(ZERO_GAS) == -6`. I figured that since this is quite close to 0, it is likely that our previous value of 2048 was a close but incorrect guess at the true zero value. I removed the reference to `ZERO_GAS` and used 0 instead.

**Verification**

Marking WIP to validate with the segments dataset as suggested by @sshane

### CODE REVIEW

**Comment by nworb-cire** [comment]
I have validated these changes on the commaCarSegments dataset, [see here](https://gist.github.com/nworb-cire/ce99a6439f03847836e1ec3b23bef656). I was a bit unsure what exactly to be looking for, but in my tests nothing looks awry. I was unable to validate a few of the platforms since they have no segments with valid carParams.

File: selfdrive/car/gm/values.py [hunk 1]
```diff
@@ -33,27 +33,26 @@ class CarControllerParams:
 
   def __init__(self, CP):
     # Gas/brake lookups
-    self.ZERO_GAS = 2048  # Coasting
```
Comment (sshane) [inline selfdrive/car/gm/values.py:36]: for future reference, this is the conversion:

`((2048 << 3) | (1 << 15) | (1 << 17)) * 0.125 - 22534`

**Comment by morrislee** [comment]
dc7716b32bf25574/00000000--b86195eae6/0
dc7716b32bf25574/00000001--a72cadb253/0
dc7716b32bf25574/00000002--df117bd704/0

**Comment by morrislee** [comment]
route with this branch and stock ACC
dc7716b32bf25574/00000004--b140f4714a/0

**Comment by Verylukyguy** [comment]
2023 Chevy Malibu in Dashcam Mode:
df51b2195de66a9b/00000000--45c2af4e69/0

2018 GMC Acadia:
df51b2195de66a9b/00000001--797163b3ba/0
df51b2195de66a9b/00000002--af4d93bcfb/0
df51b2195de66a9b/00000004--64c04ece5b/0

**Comment by morrislee** [comment]
what else do we need to have this merged? POC version currently in FrogPilot with the corrected limit (which need this PR first) makes OP Long very usable

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**